spec/determinator/retrieve/dynaconf_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'spec_helper'
+require 'determinator/retrieve/dynaconf'
+
+RSpec.describe Determinator::Retrieve::Dynaconf do
+  describe '#retrieve' do
+    let(:host) { 'DYNACONF_HOST' }
+    let(:feature_id) { 'some-feature' }
+    let(:feature_json) { {
+      name: "Feature one",
+      identifier: "feature",
+      bucket_type: "id",
+      target_groups: [{
+        rollout: 65536,
+        constraints: {}
+      }],
+      active: true,
+      overrides: {}
+    } }
+    let(:client) { instance_double(Faraday::Connection, get: nil) }
+    let(:expected_url) { "http://#{host}/scopes/florence-#{feature_id}/feature" }
+
+    shared_examples 'retrieve tests' do
+      context 'when the feature is found' do
+        let(:faraday_response) { instance_double(Faraday::Response, body: feature_json.to_json) }
+
+        before do
+          allow(client).to receive(:get).with(expected_url).and_return(faraday_response)
```

Comment:
This works, but would you consider using Webmock instead? The test would stay roughly the same, but you'd be exercising the entire implementation of the `Determinator::Retrieve::Dynaconf` object rather than stubbing its lowest layer.

I guess you could argue that it would make this more an integration test than a unit test, but it seems valuable, especially if the Faraday client starts sending the user-agent and other headers.